src/main/java/nextstep/optional/Users.java

```diff
@@ -12,12 +14,12 @@
             new User("jk", 40),
             new User("honux", 45));
 
+    // TODO 깔끔하지 않은 것 같은데 개선방안이 있을지요?
     User getUser(String name) {
-        for (User user : users) {
-            if (user.matchName(name)) {
-                return user;
-            }
-        }
-        return DEFAULT_USER;
+        Optional<User> u = users.stream().filter(
+                (user) -> user.matchName(name) ).findFirst();
+        u = Optional.ofNullable(u).orElse(Optional.of(Users.DEFAULT_USER));
+
+        return u.orElse(Users.DEFAULT_USER);
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        return users.stream()
                .filter(user -> user.matchName(name))
                .findFirst()
                .orElse(Users.DEFAULT_USER);
```